src/routes/event/details.tsx

```diff
@@ -0,0 +1,96 @@
+import { useLoaderData, useNavigate } from "react-router-dom";
+import { database, auth } from "../../firebase";
+import { get, ref } from "firebase/database";
+import { Button, Container, Stack, TextField, Typography } from "@mui/material";
+import { useEffect, useState } from "react";
+
+export async function loader({ params }) {
+  return params.eventId;
+}
+
+type EventDetailsProps = {
+  eventCategory: string;
+  eventDescription: string;
+  eventEndDate: string;
+  eventLocation: string;
+  eventOrganizer: string;
+  eventStartDate: string;
+};
+
+type EventMetadataProps = {
+  eventName: string;
+};
+
+export default function EventDetails() {
+  const eventId = useLoaderData();
+  const [eventDetails, setEventDetails] = useState<EventDetailsProps>();
+  const [eventMetadata, setEventMetadata] = useState<EventMetadataProps>({
+    eventName: "",
+  });
+  const navigate = useNavigate();
+  useEffect(() => {
+    get(ref(database, `event-details/${eventId}`)).then((snapshot) => {
+      if (snapshot.exists()) {
+        setEventDetails(snapshot.val());
+      } else {
+        console.log("No data available");
+      }
+    });
+  }, []);
+  useEffect(() => {
+    get(ref(database, `event-names/${eventId}`)).then((snapshot) => {
+      if (snapshot.exists()) {
+        setEventMetadata(snapshot.val());
+      } else {
+        console.log("No data available");
+      }
+    });
+  }, []);
```

Comment:
more API logic in the components, would love to see these moved to dedicated files just to handle this.